### ISSUE TITLE
False positive no-member for enum.Enum's _value2member_map_.

### ISSUE THREAD

**Issue by jamesbraza** [PyCQA/pylint#3941]
### Steps to reproduce

Here's a minimal repro:

```python
"""Show that pylint doesn't see enum's _value2member_map_."""

from enum import Enum


class Foo(Enum):
    """Enum class whose values we care about."""

    BAR = 0
    BAZ = 1

    @classmethod
    def has_value(cls, val: int) -> bool:
        """Get if a value is in this enum.

        SEE: https://stackoverflow.com/a/43634746/11163122

        """
        return val in cls._value2member_map_  # Location of the false positive
```

### Current behavior

Running `pylint` has the output:

```none
has_value_enum.py:19:22: E1101: Class 'Foo' has no '_value2member_map_' member (no-member)
```

### Expected behavior

As the `_value2member_map_` attribute is present, I think `pylint` should not warn that it's not present.

### pylint --version output

```none
pylint 2.6.0
astroid 2.4.2
Python 3.8.6 (default, Oct 27 2020, 11:18:42)
[Clang 12.0.0 (clang-1200.0.32.27)]
```

**Comment by hippo91** [comment]
@jamesbraza thanks for the report. It is probably a lack in the `astroid`'s `brain_namedtuple_enum` brain.

**Comment by mbyrnepr2** [comment]
> Can be closed once https://github.com/PyCQA/astroid/milestone/58 is released.

Closing!

**Comment by cdce8p** [comment]
Unfortunately, this wasn't actually included in astroid `2.12.11`. I've pinned it for `2.12.13` now.

**Comment by jacobtylerwalls** [comment]
> Unfortunately, this wasn't actually included in astroid 2.12.11.

@cdce8p can you help me understand what you're referring to? When I [compare the tags for 2.11 and 2.10](https://github.com/PyCQA/astroid/compare/v2.12.10...v2.12.11), I see this change included. I also downloaded the source for 2.11 from PyPI to be absolutely sure, and I see it there also.

We might need to undo part of https://github.com/PyCQA/astroid/pull/1863.

**Comment by cdce8p** [comment]
> > Unfortunately, this wasn't actually included in astroid 2.12.11.
> 
> @cdce8p can you help me understand what you're referring to? When I [compare the tags for 2.11 and 2.10](https://github.com/PyCQA/astroid/compare/v2.12.10...v2.12.11), I see this change included. I also downloaded the source for 2.11 from PyPI to be absolutely sure, and I see it there also.

You're right. Seems like I got confused with the different commits. Usually I open the commit from them merge message (in the PR). I.e. this one: https://github.com/PyCQA/astroid/commit/3d4dc50253a8cb0bf0ee8fda54c2bbea83bf9a92. Looking at  the branches / tags it's included only shows `main`. However that's only because the backport is a different commit.
 
> We might need to undo part of [PyCQA/astroid#1863](https://github.com/PyCQA/astroid/pull/1863).

Only partially, but yes. Would you like to do it?

**Comment by jacobtylerwalls** [comment]
Sure thing! https://github.com/PyCQA/astroid/pull/1874
